### ISSUE TITLE
Assertion generated when opening Password Safe

### ISSUE THREAD

**Issue by rochfort1** [pwsafe/pwsafe#901]
Password Safe version: v1.15 (linux local)

Operating system: Debian GNU-Linux bookworm/sid
Kernel version: 6.0.0-6-amd64
User interface: GNOME 43.1 (using Wayland)

When I attempt to open Password Safe and before I enter its password, a dialogue box opens:
The dialogue contains:
(1) The following message:
`An assertion failed!`
`./src/common/intl.cpp(423): assert ""lang != wxLANGUAGE_UNKNOWN"" failed in Init(): Initializing unknown locale doesn't make sense, did you mean to use wxLANGUAGE_DEFAULT perhaps?`
(2) A check box labelled:
**Show this dialog the next time**
(3) **Stop** and **Continue** buttons.

If I click **Continue**, the **Safe Combination Entry** dialogue box opens and access to _Password Safe_ proceeds.
If I disable **Show this dialogue the next time** and then click **Continue**, the **Safe Combination Entry** dialogue box opens and access to _Password Safe_ proceeds as before. But if I repeat this step, the assert dialogue box reappears.

I can't find `intl.cpp` to attempt an analysis of what it is trying to do. One clue might be that I have a user-defined locale country code, XA.

**Comment by ronys** [comment]
The error isn't in PasswordSafe's code, but rather in wxWidgets, the UI library that's used for Linux and Mac ports.
Specifically, https://github.com/wxWidgets/wxWidgets/blob/master/src/common/intl.cpp

According to [Wikipedia](https://en.wikipedia.org/wiki/ISO_3166-1_alpha-2#XA), 'XA' is a bit ambiguous: "The code XA is being used by Switzerland as a country code for the Canary Islands, although IC is already reserved for that purpose."

Perhaps another value, or invoking PasswordSafe via LC_ALL=C pwsafe?

**Comment by rochfort1** [comment]
@ronys Thank you for your prompt response.

I tried changing the locale to `en_XJ`. Wikipedia does not seem to think that `en_XJ` is allocated to anything (although Thunderbird reports that `en_XJ` belongs to the Jan Mayen islands). But _wxWidgets_ threw an assertion, as before. I then changed to `en_NZ` and _wxWidgets_ did not throw an assertion.

I have used the `en_XA` locale for over a year. Up until a month or so ago, wxWidgets was not throwing these assertions. At the time, I thought this was such a glaring problem that somebody would soon fix _Password Safe_. My mistake.

Now, evidently, `wxWidgets` doesn't like odd-ball locales.
